From be2e6df82979deb4675716dabf318358210c1cc9 Mon Sep 17 00:00:00 2001
From: Suren Baghdasaryan <surenb@google.com>
Date: Fri, 26 Apr 2024 20:09:37 +0000
Subject: [PATCH 5/6] ANDROID: fix !CONFIG_CMA build

MIGRATE_CMA introduced in [1] is defined only when CONFIG_CMA=y but is
used in several places even when CONFIG_CMA is not set. This breaks
builds which do not set CONFIG_CMA, like kernel_crashdump_aarch64. Add
missing CONFIG_CMA checks.

Fixes: 634c33ee9d16 ("ANDROID: mm: add cma pcp list")
Bug: 337242563
Bug: 330201547
Test: tools/bazel run --config=android_ci //common:kernel_aarch64_crashdump_dist
Change-Id: Idc5e3f236ba1fce3921fb6905f189060514ca00a
Signed-off-by: Suren Baghdasaryan <surenb@google.com>
---
 mm/page_alloc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index b33e26e64c90..60dc301d03d8 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -548,12 +548,14 @@ static inline unsigned int order_to_pindex(int migratetype, int order)
 {
 	bool __maybe_unused movable;
 
+#ifdef CONFIG_CMA
 	/*
 	 * We shouldn't get here for MIGRATE_CMA if those pages don't
 	 * have their own pcp list. For instance, free_unref_page() sets
 	 * pcpmigratetype to MIGRATE_MOVABLE.
 	 */
 	VM_BUG_ON(!cma_has_pcplist() && migratetype == MIGRATE_CMA);
+#endif
 
 #ifdef CONFIG_TRANSPARENT_HUGEPAGE
 	if (order > PAGE_ALLOC_COSTLY_ORDER) {
@@ -2758,7 +2760,9 @@ void free_unref_page(struct page *page, unsigned int order)
 			free_one_page(page_zone(page), page, pfn, order, FPI_NONE);
 			return;
 		}
+#ifdef CONFIG_CMA
 		if (!cma_has_pcplist() || migratetype != MIGRATE_CMA)
+#endif
 			migratetype = MIGRATE_MOVABLE;
 	}
 
@@ -2858,7 +2862,9 @@ void free_unref_folios(struct folio_batch *folios)
 		 * to the MIGRATE_MOVABLE pcp list.
 		 */
 		if (unlikely(migratetype > MIGRATE_RECLAIMABLE)) {
+#ifdef CONFIG_CMA
 			if (!cma_has_pcplist() || migratetype != MIGRATE_CMA)
+#endif
 				migratetype = MIGRATE_MOVABLE;
 		}
 
-- 
2.43.0

